packages/components/addon/components/hds/copy/button/index.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export const DEFAULT_SIZE = 'medium';
+export const SIZES = ['small', 'medium'];
+export const DEFAULT_ICON = 'clipboard-copy';
+export const SUCCESS_ICON = 'clipboard-checked';
+
+export default class HdsCopyButtonComponent extends Component {
+  @tracked isSuccess = false;
+  @tracked isError = false;
+  @tracked icon = DEFAULT_ICON;
+
+  /**
+   * @param clipboardText
+   * @type {string}
+   * @description The ID of the element containing the text to be copied. If no clipboardText value is defined `isError` will be set to true.
+   */
+  get clipboardText() {
```

Comment:
I think there is some confusion between `@clipboardText` and `@clipboardTarget` (see the implementation in Cloud UI - [@clipboardText](https://github.com/search?q=repo%3Ahashicorp%2Fcloud-ui%20%40clipboardText&type=code) and [@clipboardTarget](https://github.com/search?q=repo%3Ahashicorp%2Fcloud-ui%20%40clipboardTarget&type=code) - that mimics the underlying implementation in [ember-cli-clipboard](https://github.com/jkusa/ember-cli-clipboard) / [clipboard.js](https://github.com/zenorocha/clipboard.js) where
- `@clipboardText` it's used to copy text provided as an argument (not associated to any DOM element, just in memory)
- `@clipboardTarget` it's used to copy text from a DOM node, where the "target" is a path selector for that node

In our implementation, or we support only the first one, or we support both. We can't support only the second one, because it doesn't work with `<input/textarea>` elements (see comment below)